internal/infrastructure/kubernetes/infra_resource.go

```diff
@@ -73,6 +73,14 @@ func (i *Infra) createOrUpdateDeployment(ctx context.Context, r ResourceRender)
 	}
 
 	return i.Client.CreateOrUpdate(ctx, key, current, deployment, func() bool {
+		// applied to k8s the "DeprecatedServiceAccount" will fill it.
+		deployment.Spec.Template.Spec.DeprecatedServiceAccount = current.Spec.Template.Spec.DeprecatedServiceAccount
+		// applied to k8s the "SecurityContext" will fill it with default settings.
+		deployment.Spec.Template.Spec.SecurityContext = current.Spec.Template.Spec.SecurityContext
+		// adapter the hpa updating and envoyproxy updating.
+		if *deployment.Spec.Replicas < *current.Spec.Replicas {
```

Comment:
thanks for raising this issue, imo we should either
* skip comparing replica if unset (lets make sure its a pointer in the API) and highlight in HPA docs to do so
* if replica is set in our API, we recommend not using HPA, and use that value for comparison

Reply:
> skip comparing replica if unset (lets make sure its a pointer in the API) and highlight in HPA docs to do so

unset the k8s will fill the default value "1", then `current` `replicas` is not equal to `deployment` will fire an update.

> if the replica is set in our API, we recommend not using HPA, and using that value for comparison

IMO, we can compare the replicas from envoyporoxy or k8s like the current compare